lib/oban.ex

```diff
@@ -840,4 +861,16 @@ defmodule Oban do
 
     with {:error, error} <- conf.engine.init(conf, opts), do: raise(error)
   end
+
+  defp maybe_validate_queue_exists!(name, opts) do
+    if Keyword.get(opts, :local_only) do
+      validate_queue_exists!(name, opts[:queue])
+    end
+  end
+
+  defp validate_queue_exists!(name, queue) do
```

Comment:
It's ok to use a single function and omit the `maybe_` prefixed one.

Reply:
Just gave this a shot, I ended up going for an `if` despite breaking consistency with the `unless ..., do: raise ...` elsewhere because it was a little hard to grok at a glance.

Let me know if you'd prefer this in some other way!